src/Processors/Formats/InputFormatErrorsLogger.cpp

```diff
@@ -0,0 +1,81 @@
+#include <Processors/Formats/InputFormatErrorsLogger.h>
+#include <DataTypes/DataTypeDateTime.h>
+#include <DataTypes/DataTypeNullable.h>
+#include <DataTypes/DataTypeString.h>
+#include <DataTypes/DataTypesNumber.h>
+#include <IO/WriteHelpers.h>
+#include <Processors/Formats/IRowOutputFormat.h>
+
+
+namespace DB
+{
+
+namespace
+{
+    const String DEFAULT_OUTPUT_FORMAT = "CSV";
+}
+
+InputFormatErrorsLogger::InputFormatErrorsLogger(const ContextPtr & context)
+{
+    String output_format = context->getSettingsRef().errors_output_format;
+    if (!FormatFactory::instance().isOutputFormat(output_format))
+        output_format = DEFAULT_OUTPUT_FORMAT;
+    try
+    {
+        table = context->getInsertionTable().getTableName();
+        database = context->getInsertionTable().getDatabaseName();
+    }
+    catch (...)
+    {
+        /// Ignore
+    }
```

Comment:
```suggestion
    if (context->hasInsertionTable())
    {
        table = context->getInsertionTable().getTableName();
        database = context->getInsertionTable().getDatabaseName();
    }
```

Reply:
But If the database is empty, it will throw an exception.

Reply:
I see. As I understand, it can happen only in client side because we set empty database when insert query doesn't contain it. We should use default database in this case instead of empty one (and to be more precise, we should use `config().getString("database", "default") in client, because we can specify database in client parameters`)

Reply:
What about inserting into a temporary table? (names of temporary tables are never qualified with a database name)

Reply:
Ah, right, forgot about it. Then yes, we can have cases when database is empty. Then it would be better to add a new method `bool hasDatabase()` in `StorageID` and use it here instead of `try/catch`.

Reply:
Thanks, I get it.